File: backend/node-backend/routes/analytics.js
/**
 * Analytics Routes
 * ----------------
 * Endpoints for logging and retrieving analytics events.
 */

const express = require('express');
const router = express.Router();

// Temporary in‑memory store for testing
let analytics = [];

// Log event (no auth for now)
router.post('/', (req, res) => {
  const { text, label, confidence, timestamp } = req.body;

  if (!text || !label) {
    return res.status(400).json({ error: "Missing required fields" });
  }

  const entry = { text, label, confidence, timestamp };
  analytics.push(entry);

  res.json({ message: "Analytics logged", entry });
});

// Get logs (no auth for now)
router.get('/', (req, res) => {
  res.json(analytics);
});

module.exports = router;

File: BACKEND/ml-service/app/routers/analytics.py
from fastapi import APIRouter
from pydantic import BaseModel
from datetime import datetime

router = APIRouter()

class ModerationLog(BaseModel):
    text: str
    label: str
    confidence: float
    timestamp: datetime

# Temporary in-memory store
logs = []

@router.post("/ml/analytics")
def log_moderation(data: ModerationLog):
    logs.append(data)
    return {"message": "Log stored", "total_logs": len(logs)}

@router.get("/ml/analytics")
def get_logs():
    return {"logs": logs}
